Apps/W1/ContosoCoffeeDemoDataset/app/Service/3.Transactions/CreateSvcDemoTransactions.Codeunit.al

```diff
@@ -0,0 +1,195 @@
+        ItemJournalTemplate.SetRange(Recurring, false);
+        ItemJournalTemplate.SetRange(Type, ItemJournalTemplate.Type::Item);
+
+        if ItemJournalTemplate.FindFirst() then
+            if not ItemJournalBatch.Get(ItemJournalTemplate.Name, ItemJnlBatchName) then begin
+                ItemJournalBatch.Init();
+                ItemJournalBatch."Journal Template Name" := ItemJournalTemplate.Name;
+                ItemJournalBatch.SetupNewBatch();
+                ItemJournalBatch.Name := ItemJnlBatchName;
+                ItemJournalBatch.Description := ItemJnlBatchName;
+                ItemJournalBatch.Insert(true);
+                Commit();
+            end;
+        ItemJnlTemplateName := ItemJournalTemplate.Name;
+    end;
+
+    local procedure InitItemJnlLine(var ItemJournalLine: Record "Item Journal Line"; JournalTemplateName: Code[10]; JournalBatchName: Code[10])
+    var
+        LastItemJournalLine: Record "Item Journal Line";
+    begin
+        ItemJournalLine.Init();
+        ItemJournalLine.Validate("Journal Template Name", JournalTemplateName);
+        ItemJournalLine.Validate("Journal Batch Name", JournalBatchName);
+
+        LastItemJournalLine.Reset();
+        LastItemJournalLine.SetRange("Journal Template Name", JournalTemplateName);
+        LastItemJournalLine.SetRange("Journal Batch Name", JournalBatchName);
+        if LastItemJournalLine.FindLast() then
+            LineNumber := LastItemJournalLine."Line No." + 10000
+        else
+            LineNumber := 10000;
+        ItemJournalLine.Validate("Line No.", LineNumber);
+    end;
+
+    local procedure CreateLoaners()
+    var
+        Loaner: Record "Loaner";
+    begin
+        if not Loaner.Get(XLOANER1Tok) then begin
+            Loaner.Init();
+            Loaner.Validate("No.", XLOANER1Tok);
+            Loaner.Validate("Description", XLOANER1Tok);
+            Loaner.Validate("Item No.", SvcDemoDataSetup."Item 1 No.");
+            OnBeforeLoanerInsert(Loaner);
+            Loaner.Insert(DoRunTriggers);
+        end;
+        if not Loaner.Get(XLOANER2Tok) then begin
+            Loaner.Init();
+            Loaner.Validate("No.", XLOANER2Tok);
+            Loaner.Validate("Description", XLOANER2Tok);
+            Loaner.Validate("Item No.", SvcDemoDataSetup."Item 2 No.");
+            OnBeforeLoanerInsert(Loaner);
+            Loaner.Insert(DoRunTriggers);
+        end;
+    end;
+
+    local procedure CreateSalesOrders()
+    begin
+        CreateSalesOrder(SvcDemoDataSetup."Customer No.", SvcDemoDataSetup."Item 1 No.", 'SVC-1');
+        CreateSalesOrder(SvcDemoDataSetup."Customer No.", SvcDemoDataSetup."Item 2 No.", 'SVC-2');
+    end;
+
+    local procedure CreateSalesOrder(CustomerNo: Code[20]; ItemNo: Code[20]; ExternalDocumentNo: Code[20])
+    var
+        SalesHeader: Record "Sales Header";
+        SalesLine: Record "Sales Line";
+    begin
+        SalesHeader.Init();
```

Comment:
is this by design? we insert first and modify later

Reply:
I mean - I can try it without that - it's old habit, since the Insert(true) gives it a Document Number, then we run into less issues (historically) with validation of fields.   Can't recall how old that habit is at this point, so it might be not necessary these days.